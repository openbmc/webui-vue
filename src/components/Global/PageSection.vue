<template>
  <div class="page-section">
    <h2 v-if="sectionTitle">{{ sectionTitle }}</h2>
    <slot />
  </div>
</template>

<script>
export default {
  name: 'PageSection',
  props: {
    sectionTitle: {
      type: String,
      default: '',
    },
  },
};
</script>

<style lang="scss" scoped>
.page-section {
  margin-bottom: $spacer * 4;
}

h2 {
  @include font-size($h3-font-size);
  margin-bottom: $spacer;
}
</style>
