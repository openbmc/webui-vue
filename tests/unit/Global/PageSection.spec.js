import { mount, createLocalVue } from '@vue/test-utils';
import PageSection from '@/components/Global/PageSection';

const localVue = createLocalVue();

describe('PageSection.vue', () => {
  const wrapper = mount(PageSection, {
    localVue,
    propsData: {
      sectionTitle: 'PageSection test title',
    },
    mocks: {
      $t: (key) => key,
    },
  });
  it('should exist', () => {
    expect(wrapper.exists()).toBe(true);
  });
  it('should render h2 element', () => {
    expect(wrapper.find('h2').exists()).toBe(true);
  });
  it('should render correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
