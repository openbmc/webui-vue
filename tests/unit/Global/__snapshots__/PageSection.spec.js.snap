// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSection.vue should render correctly 1`] = `
<div
  class="page-section"
>
  <h2>
    PageSection test title
  </h2>
   
</div>
`;
